services/flake_detection.py

```diff
@@ -119,19 +120,26 @@ def __init__(self):
 
 
 class UnrelatedMatchesDetector(BaseSymptomDetector):
-    def __init__(self):
+    def __init__(self, failure_normalizer=None):
         self.state = defaultdict(lambda: defaultdict(TestDictObject))
         self.res = defaultdict(list)
+        self.failure_normalizer: FailureNormalizer | None = failure_normalizer
```

Comment:
Why put the type annotation in the attribute instead of on the function argument?
The attribute type can be derived from the argument, not the other way around.

Reply:
very good point